### PR TITLE
Dx 245 ambient temp

### PR DESCRIPTION
**Pull request by rlperez**
Add ambient temperature state

### CODE REVIEW

**Review by ses110 — APPROVED** [review]
Looks good to me 👍

File: doc/api/yonomi-sdk/ThermostatRepository/getThermostatTrait.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ static ThermostatTrait getThermostatTrait(dynamic trait) {
       FanMode(trait.state.fanMode.reported?.value ?? AvailableFanMode.ON),
       ThermostatMode(
           trait.state.mode.reported?.value ?? AvailableThermostatMode.OFF),
+      AmbientTemperature(
+          trait.state.ambientTemperature.reported?.value ?? 0.0),
```
Comment (grizzly-atoms) [inline doc/api/yonomi-sdk/ThermostatRepository/getThermostatTrait.md:44]: Should this default to 0.0? Seems like we might want it to default to null, as 0.0 means something pretty specific.

File: lib/src/repository/traits/thermostat_repository.dart [hunk 2]
```diff
@@ -30,6 +30,8 @@ class ThermostatRepository {
         FanMode(trait.state.fanMode.reported?.value ?? AvailableFanMode.ON),
         ThermostatMode(
             trait.state.mode.reported?.value ?? AvailableThermostatMode.OFF),
+        AmbientTemperature(
+            trait.state.ambientTemperature.reported?.value ?? 0.0),
```
Comment (grizzly-atoms) [inline lib/src/repository/traits/thermostat_repository.dart:34]: default to null???

**Review by grizzly-atoms — CHANGES_REQUESTED** [review]
Let's chat with Alex, but I'm pretty sure we shouldn't default temp values to any numerical value.  These should probably show up as null if for some reason it's not available.

File: lib/src/repository/traits/thermostat_repository.dart (hunk 2 above)
Comment (rlperez) [reply lib/src/repository/traits/thermostat_repository.dart:34]: If this is the case would we want to include the state at all in the trait?

Comment (rlperez) [reply lib/src/repository/traits/thermostat_repository.dart:34]: If we didn't include the state then we would probably want to make it's argument non nullable and then we would only handle the states that are present.

**Comment by ses110** [comment]
> Let's chat with Alex, but I'm pretty sure we shouldn't default temp values to any numerical value. These should probably show up as null if for some reason it's not available.

This is a good discussion topic. We have to consider the data types carefully at this level since this SDK is more concerned about parsing and encapsulating device data directly from the platform but not concerned about what shows up at the UI level.

For example, we should agree on whether to use nullable variables or not to represent things like Temperature set point. Once we agree on the approach, we can cascade null values safely to the `device-widgets` SDK and handle those null values in a general way.

**Review by grizzly-atoms — APPROVED** [review]

**Pull request merged**